test/solc/CommandLineInterfaceAllowPaths.cpp

```diff
@@ -508,6 +514,37 @@ BOOST_FIXTURE_TEST_CASE(allow_path_automatic_whitelisting_work_dir, AllowPathsFi
 	BOOST_TEST(checkImport("import 'a/X/c.sol'", {"--base-path", ""}));
 }
 
+BOOST_FIXTURE_TEST_CASE(allow_path_automatic_whitelisting_include_paths, AllowPathsFixture)
+{
+	// Relative include path whitelists its content
+	BOOST_TEST(checkImport("import 'b/c.sol'", {"--base-path=a/b/c", "--include-path=../code/a"}));
```

Comment:
I don't really understand what these are testing.

Reply:
Overall these tests check if this particular source file (the import is the only thing in it) compiled with these specific command-line options compiles fine or if it reports some kind of error (path not allowed or file missing). It's just wrapped in a helper function to allow me to test a lot of such combinations. There are tons of weird corner cases related to allowed paths because the feature interacts with all the other path-related options.

You might wan to look at `AllowPathsFixture` to see a list of files which are automatically created by test setup. Basically, there's a `work/` directory that is the current working directory when the compilation is executed and `code/`, which is specifically meant to be outside of the work dir. Both contain `a/b/c.sol`.

This particular test adds `code/a/` as an include path and `work/a/b/c/` as the base path. Then it checks importing `b/c.sol` will make the compiler find `code/a/b/c.sol` instead of complaining that the path is disallowed or that the file does not exist.